File: 1890-sum-of-beauty-of-all-substrings/1890-sum-of-beauty-of-all-substrings.java
class Solution {
    public int beautySum(String s) {
        int n = s.length();
        int totalBeauty = 0;
        
        for (int i = 0; i < n; i++) {
            int[] hash = new int[26];
            for (int j = i; j < n; j++) {
                hash[s.charAt(j) - 'a']++;
                int max = Integer.MIN_VALUE, min = Integer.MAX_VALUE;
                for (int k = 0; k < 26; k++) {
                    if (hash[k] > 0) {
                        max = Math.max(max, hash[k]);
                        min = Math.min(min, hash[k]);
                    }
                }
                totalBeauty += max - min;
            }
        }
        
        return totalBeauty;
    }
}

File: 1890-sum-of-beauty-of-all-substrings/1890-sum-of-beauty-of-all-substrings.py
class Solution(object):
    def beautySum(self, s):
        n=len(s)
        total_beauty=0

        for i in range(n):
            freq=[0]*26
            for j in range(i,n):
                freq[ord(s[j])-ord('a')]+=1
                max_value = -(2**31)  
                min_value = (2**31)-1
                for k in range(26):
                    if freq[k]>0:
                        max_value=max(max_value,freq[k])
                        min_value=min(min_value,freq[k])
                total_beauty+=max_value-min_value

        return total_beauty